{
	"id": "CVE-2018-19351",
	"vulnerabilities": [
		{
			"content": {
				"id": "CVE-2018-19351",
				"description": "Jupyter Notebook before 5.7.1 allows XSS via an untrusted notebook because nbconvert responses are considered to have the same origin as the notebook server. In other words, nbconvert endpoints can execute JavaScript with access to the server API. In notebook/nbconvert/handlers.py, NbconvertFileHandler and NbconvertPostHandler do not set a Content Security Policy to prevent this.",
				"severity": [
					{
						"type": "cvss_v2",
						"source": "nvd@nist.gov",
						"cvss_v2": {
							"vector": "AV:N/AC:M/Au:N/C:N/I:P/A:N",
							"base_score": 4.3,
							"nvd_base_severity": "MEDIUM",
							"temporal_score": 4.3,
							"nvd_temporal_severity": "MEDIUM",
							"environmental_score": 4.3,
							"nvd_environmental_severity": "MEDIUM"
						}
					},
					{
						"type": "cvss_v30",
						"source": "nvd@nist.gov",
						"cvss_v30": {
							"vector": "CVSS:3.0/AV:N/AC:L/PR:N/UI:R/S:C/C:L/I:L/A:N",
							"base_score": 6.1,
							"base_severity": "MEDIUM",
							"temporal_score": 6.1,
							"temporal_severity": "MEDIUM",
							"environmental_score": 6.1,
							"environmental_severity": "MEDIUM"
						}
					}
				],
				"cwe": [
					{
						"source": "nvd@nist.gov",
						"cwe": [
							"CWE-79"
						]
					}
				],
				"references": [
					{
						"source": "cve@mitre.org",
						"url": "https://github.com/jupyter/notebook/blob/master/docs/source/changelog.rst"
					},
					{
						"source": "cve@mitre.org",
						"url": "https://github.com/jupyter/notebook/commit/107a89fce5f413fb5728c1c5d2c7788e1fb17491"
					},
					{
						"source": "cve@mitre.org",
						"url": "https://groups.google.com/forum/#%21topic/jupyter/hWzu2BSsplY"
					},
					{
						"source": "cve@mitre.org",
						"url": "https://lists.debian.org/debian-lts-announce/2020/11/msg00033.html"
					},
					{
						"source": "cve@mitre.org",
						"url": "https://pypi.org/project/notebook/#history"
					},
					{
						"source": "nvd.nist.gov",
						"url": "https://nvd.nist.gov/vuln/detail/CVE-2018-19351"
					}
				],
				"published": "2018-11-18T17:29:00.267Z",
				"modified": "2023-11-07T02:55:31.847Z"
			},
			"segments": [
				{
					"ecosystem": "cpe"
				}
			]
		}
	],
	"detections": [
		{
			"ecosystem": "cpe",
			"conditions": [
				{
					"criteria": {
						"operator": "OR",
						"criterias": [
							{
								"operator": "OR",
								"criterias": [
									{
										"operator": "OR",
										"criterias": [
											{
												"operator": "OR",
												"criterions": [
													{
														"vulnerable": true,
														"fix_status": {
															"class": "unknown"
														},
														"package": {
															"cpe": "cpe:2.3:a:jupyter:notebook:*:*:*:*:*:*:*:*"
														},
														"affected": {
															"type": "semver",
															"range": [
																{
																	"lt": "5.7.1"
																}
															]
														}
													}
												]
											}
										]
									}
								]
							}
						]
					}
				}
			]
		}
	],
	"data_source": {
		"id": "nvd-api-cve",
		"raws": [
			"vuls-data-raw-nvd-api-cve/2018/CVE-2018-19351.json"
		]
	}
}
